pkg/apis/apps/v1alpha1/servicebindingrequest_types.go

```diff
@@ -21,76 +21,40 @@ type ServiceBindingRequestSpec struct {
 	EnvVarPrefix string `json:"envVarPrefix,omitempty"`
 
 	// BackingServiceSelector is used to identify the backing service operator.
-	//
-	// Refer: https://12factor.net/backing-services
-	// A backing service is any service the app consumes over the network as
-	// part of its normal operation. Examples include datastores (such as
-	// MySQL or CouchDB), messaging/queueing systems (such as RabbitMQ or
-	// Beanstalkd), SMTP services for outbound email (such as Postfix), and
-	// caching systems (such as Memcached).
-	//
-	// Example 1:
-	//	backingServiceSelector:
-	//		resourceKind: databases.example.org
-	//      resourceRef: mysql-database
-	// Example 2:
-	//	backingServiceSelector:
-	//		resourceKind: databases.example.org
-	//		resourceVersion: v1alpha1
-	//      resourceRef: mysql-database
 	BackingServiceSelector BackingServiceSelector `json:"backingServiceSelector"`
 
 	// ApplicationSelector is used to identify the application connecting to the
 	// backing service operator.
-	// Example 1:
-	//	applicationSelector:
-	//		matchLabels:
-	//			connects-to: postgres
-	//			environment: stage
-	//		resourceKind: Deployment
-	// Example 2:
-	//	applicationSelector:
-	//		matchLabels:
-	//			connects-to: postgres
-	//			environment: stage
 	ApplicationSelector ApplicationSelector `json:"applicationSelector"`
 }
 
+// ServiceBindingRequestStatus defines the observed state of ServiceBindingRequest
+// +k8s:openapi-gen=true
+type ServiceBindingRequestStatus struct {
+	// BindingStatus is the status of the service binding request.
+	BindingStatus string `json:"bindingStatus,omitempty"`
+	// Secret is the name of the intermediate secret
+	Secret string `json:"secret,omitempty"`
+	// ApplicationObjects contains all the application objects filtered by label
+	ApplicationObjects []string `json:"applicationObjects,omitempty"`
+}
+
 // BackingServiceSelector defines the selector based on resource name, version, and resource kind
 // +k8s:openapi-gen=true
 type BackingServiceSelector struct {
-	Group       *string `json:"group,omitempty"`
-	Version     string  `json:"version"`
-	Kind        string  `json:"kind"`
-	ResourceRef string  `json:"resourceRef"`
+	Group       string `json:"group,omitempty"`
+	Version     string `json:"version"`
+	Kind        string `json:"kind"`
```

Comment:
You could replace this with `Resource` here as well, since you are going to need it for https://github.com/sbose78/service-binding-operator/pull/2/files#diff-42835fd3b25b196b1bbc21495e0ac5b7R149 . I  could do in a different PR as well - as you wish :)

Reply:
@sbose78 please check the approach in `planner.go` in order to exchange a GVK for a GVR. Both scenarios, on going from GVK to GVR or vice-versa would be the same effort. (cc @isutton)

Reply:
We are going to need the `Resource for `crResource := schema.GroupVersionResource{plan.CR.GroupVersionKind().Group, plan.CR.GroupVersionKind().Version,"databases"` 

`Kind` unfortunately doesn't help. But we can skip making that change in this PR as it is not needed here :)

Reply:
Agreed. It's a good idea making this change when we are touching that part.